web/groups.go

```diff
@@ -197,14 +198,21 @@ func (s *QuickFeedService) getGroupUsers(request *qf.Group) ([]*qf.User, error)
 	return users, nil
 }
 
-// only allow letters, numbers, dash and underscore.
-var regexpNonAuthorizedChars = regexp.MustCompile("[^a-zA-Z0-9-_]")
+// only allow letters, numbers, underscore, dash and period.
+var regexpNonAuthorizedChars = regexp.MustCompile("[^a-zA-Z0-9_.-]")
 
 // checkGroupName returns an error if the group name is invalid; otherwise nil is returned.
 func (s *QuickFeedService) checkGroupName(courseID uint64, groupName string) error {
+	if groupName == "" {
+		return ErrGroupNameEmpty
+	}
 	if len(groupName) > maxGroupNameLength {
 		return ErrGroupNameTooLong
 	}
+	// check if group name contains space(s)
```

Comment:
Missing test for this. Also,  don’t think it is needed as a separate check since the reg exp does not allow it.